### ISSUE TITLE
bug: SideNav hover color contrast accessability issue

### ISSUE THREAD

**Issue by s-laugh** [cds-snc/gcds-components#1143]
### Prerequisites | Prérequis

- [x] I have read the [Contributing Guidelines](https://github.com/cds-snc/gcds-components/blob/main/CONTRIBUTING.md). | J’ai lu les [Lignes directrices sur les contributions](https://github.com/cds-snc/gcds-components/blob/main/CONTRIBUTING.md).
- [x] I agree to follow the [Code of Conduct](https://github.com/cds-snc/gcds-components/blob/main/CODE_OF_CONDUCT.md). | J’accepte de me conformer au [Code de conduite](https://github.com/cds-snc/gcds-components/blob/main/CODE_OF_CONDUCT.md).
- [x] I have searched for [existing issues](https://github.com/cds-snc/gcds-components/issues) that already report this problem, without success. | J’ai vérifié parmi [les tickets (issues) existants](https://github.com/cds-snc/gcds-components/issues) que ce sujet n’avait pas déjà été soumis.

### GC Design System Components Package and Version | Paquet et version des composants de Système de design GC

gcds-components@1.0.0

### Current Behavior | Comportement observé

Side nav hover color #f1f2f3 on the standard white background #ffffff doesn't meet WCAG contrast ratio of 3:1 for meaningful non-text elements.

### Expected Behavior | Comportement attendu

All WCAG guidelines are met.

### System Info | Information sur le système

```shell

```

### Steps to Reproduce | Étapes pour reproduire le bogue

```html
<gcds-side-nav>
    <gcds-nav-link>
        Lifetime
    </gcds-nav-link>
</gcds-side-nav>
```

### Additional Information | Informations supplémentaires

https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast.html

Was reported to my team by ESDC a11y audit https://esdc.prv/cgi-bin/scdtic-ictdcs/page-error-view.aspx?id=37325&wid=0
